tests/proofs/dexter/dexter.md

```diff
@@ -201,6 +208,139 @@ As reference materials for understanding the contract intent, we will consult:
 4.  The [FA 1.2 standard](https://gitlab.com/tzip/tzip/blob/master/proposals/tzip-7/tzip-7.md)
 5.  The [FA 2 standard](https://gitlab.com/tzip/tzip/-/blob/master/proposals/tzip-12/tzip-12.md)
 
+## State Abstraction
+
+We use a few helper routines to convert between our abstract and concrete proof state.
+We first define functions which build our parameter and our storage types.
+
+```k
+  syntax TypeName ::= #DexterParamType(Bool)                [function, functional]
+                    | #DexterVersionSpecificParamType(Bool) [function, functional]
+  // -----------------------------------------------------------------------------
+  rule #DexterParamType(IsFA2)
+    => or
+         or
+           or
+             or pair address                         // addLiquidity
+                  pair nat
+                    pair nat timestamp
+                unit                                 // default
+             or pair address                         // removeLiquidity
+                  pair nat
+                    pair mutez
+                      pair nat timestamp
+                pair option key_hash bool            // setBaker
+           or
+             or address                              // setLqtAddress
+                address                              // setManager
+             or pair address                         // tokenToToken
+                  pair nat
+                    pair address
+                      pair nat timestamp
+                pair address                         // tokenToXtz
+                  pair nat
+                    pair mutez timestamp
+         or
+           or unit                                   // updateTokenPool
+              #DexterVersionSpecificParamType(IsFA2) // updateTokenPoolInternal
+           pair address                              // xtzToToken
+             pair nat timestamp
```

Comment:
Could we use parentheses for these constructs? I assume `pair` binds tighter than `or`, and they both associate to the right, but I'd rather the bindings be explicit.

Reply:
I'd say it's not necessary for the less complex types (just tuples), but I find it easier to navigate a complex type like this with parentheses. It also makes it easier for my editor to help me with that.

Reply:
I added parentheses around the `or` types constructors, but left the nested pairs as is, because I find that too many parens can obscure meaning. 

BTW, the reason I avoided adding extra parens is because of some bad experiences fighting with the parser when writing a scheme-like like language semantics. However, in the context of Michelson, I don't think I need to be so vigilant.